src/main/java/org/galatea/starter/entrypoint/IexRestController.java

```diff
@@ -48,4 +49,30 @@
     return iexService.getLastTradedPriceForSymbols(symbols);
   }
 
+  /**
+   * Get the historical price data for a given stock symbol.
+   *
+   * @param symbol stock symbol to get historical data for.
+   * @param range specified time range of the request.
+   * @param date specified date in the format YYYYMMDD.
+   * @return a list of historical price data for the given symbol, range, and date.
+   */
+  @GetMapping(value = "${mvc.iex.getHistoricalPricePath}", produces = {
+      MediaType.APPLICATION_JSON_VALUE})
+  public List<IexHistoricalPrice> getHistoricalPrice(
+      @RequestParam(value = "symbol") final String symbol,
+      @RequestParam(value = "range", required = false) final String range,
+      @RequestParam(value = "date", required = false) final String date) {
+    if (date == null && range == null) {
+      return iexService.getHistoricalPriceForSymbol(symbol);
+    }
+    if (date == null) {
+      return iexService.getHistoricalPriceByRange(symbol, range);
+    }
+    if (range == null) {
+      return iexService.getHistoricalPriceByDate(symbol, date);
+    }
+    return iexService.getHistoricalPriceByDate(symbol, range, date);
+  }
+
```

Comment:
Some people would consolidate this into a nested conditional, but it's purely a stylistical change. I find it makes it just a little easier to quickly look and see which method will be called

Reply:
I tried my best to make the code cleaner with nested conditionals in IexService, but please let me know if the current structure seems off.